test/nettests/dash.cpp

```diff
@@ -15,8 +15,4 @@ TEST_CASE("Synchronous dash test") {
     test::nettests::make_test<DashTest>().run();
 }
 
-TEST_CASE("Asynchronous dash test") {
-    test::nettests::run_async(test::nettests::make_test<DashTest>());
-}
-
```

Comment:
I have removed async execution of tests since now we use the same code for async and non-async execution. There may be a little penalty for doing sync execution in another thread, but our main use case is _really_ the case of async execution in a background thread. As such, this does not worry me much that we're implementing sync execution using async execution plus a future object. Removing async execution for each test, given that it's now redundant, can significantly reduce the time it takes to run all tests on travis, so I see it as a plus.